src/javascript/binary/pages/trade/message.js

```diff
@@ -3,33 +3,97 @@
  * events based on type of response
  */
 var Message = (function () {
-    'use strict';
+    "use strict";
 
     var process = function (msg) {
-        var response = JSON.parse(msg.data);
-        if (response) {
-            var type = response.msg_type;
-            if (type === 'authorize') {
-                TUser.set(response.authorize);
-                TradeSocket.send({ payout_currencies: 1 });
-            } else if (type === 'active_symbols') {
+
+        var response;
+
+        try {
+            response  = JSON.parse(msg.data);
+            if("object" !== typeof response || !("msg_type" in response)) {
+                throw new Error("Response from WS API is not well formatted.");
+            }
+        } catch(e) {
+            throw new Error("Response from WS API is not well formatted.");
```

Comment:
The thinking here has been, (1) the response to API request might be be totally screwed and not a valid JSON at all. If that is the case, `response  = JSON.parse(msg.data);` will cause a Fatal error (hence the `try` block and the throw in the catch block), or (2) it is valid JSON and we succeed in parsing it into an object, but there is no `msg_type` key in it. The code that follows relies on this key to exist, if it does not, we again complain that the response is not well formatted and terminate further execution.

But anyway, if this is being paranoid, and we don't need these checks, I'll remove them.